src/prefect/orion/models/workers_migration.py

```diff
@@ -0,0 +1,171 @@
+    )
+    if pool is None:
+        pool = await models.workers.create_work_pool(
+            session=session,
+            work_pool=schemas.actions.WorkPoolCreate(
+                name=DEFAULT_AGENT_WORK_POOL_NAME,
+                description="A default work pool for Prefect Agents",
+                type=PREFECT_AGENT_WORK_POOL_TYPE,
+            ),
+        )
+    return pool
+
+
+@inject_db
+async def get_or_create_work_pool_queue(
+    session: AsyncSession,
+    work_queue_id: Optional[UUID] = None,
+    work_queue_name: Optional[str] = None,
+    db: OrionDBInterface = None,
+):
+    """
+    Gets or creates a work pool queue for a work queue.
+
+    Will update corresponding deployments and flow runs to point at the new work pool queue.
+
+    Args:
+        session (AsyncSession): a database session
+        work_queue_id (UUID): the work queue id
+        work_queue_name (str): the work queue name
+        db (OrionDBInterface, optional): a database interface. Defaults to None.
+
+    Returns:
+        db.WorkPoolQueue: The work pool queue
+    """
+
+    if (work_queue_id, work_queue_name) == (None, None):
+        raise ValueError(
+            "Either the work queue name or the work queue ID must be provided."
+        )
+
+    if work_queue_id is not None:
+        work_queue = await models.work_queues.read_work_queue(
+            session=session, work_queue_id=work_queue_id
+        )
+    else:
+        work_queue = await models.work_queues.read_work_queue_by_name(
+            session=session, name=work_queue_name
+        )
+
+    if not work_queue:
+        raise ObjectNotFoundError()
+
+    # get or create the default agents pool
+    pool = await get_or_create_default_agent_work_pool(session=session, db=db)
+
+    # get or create the work pool queue
+    work_pool_queue = await models.workers.read_work_pool_queue_by_name(
+        session=session,
+        work_pool_name=DEFAULT_AGENT_WORK_POOL_NAME,
+        work_pool_queue_name=work_queue.name,
+    )
+    if work_pool_queue is None:
+        work_pool_queue = await models.workers.create_work_pool_queue(
+            session=session,
+            work_pool_id=pool.id,
+            work_pool_queue=schemas.actions.WorkPoolQueueCreate(
+                name=work_queue.name,
+                description=work_queue.description,
+                is_paused=work_queue.is_paused,
+                concurrency_limit=work_queue.concurrency_limit,
+            ),
+        )
+
+        # update deployments to point at the new work queue
+        update_deployments_stmt = (
+            sa.update(db.Deployment)
+            .where(
+                db.Deployment.work_queue_name == work_queue.name,
+                db.Deployment.work_pool_queue_id.is_(None),
+            )
+            .values(work_pool_queue_id=work_pool_queue.id)
+        )
+        await session.execute(update_deployments_stmt)
+
+        # update scheduled flow runs to point at the new work queue
```

Comment:
We should add some batching here, updating all flow runs at once will cause timeouts.

Can we `UPDATE WHERE id in (SELECT id from flow_run where <WHERE CLAUSE>` and COMMIT periodically?